api/admin/service.go

```diff
@@ -83,9 +85,22 @@ func NewService(config Config) (*common.HTTPHandler, error) {
 	}, "admin"); err != nil {
 		return nil, err
 	}
+
+	// Check if the given hostname matches the configured host via CLI
+	newServer.RegisterValidateRequestFunc(BlockRequestsWithNonMatchingHostnames(config.HTTPHost))
+
 	return &common.HTTPHandler{Handler: newServer}, nil
 }
 
+func BlockRequestsWithNonMatchingHostnames(hostName string) func(r *rpc.RequestInfo, _i interface{}) error {
+	return func(r *rpc.RequestInfo, _i interface{}) error {
+		if r.Request.Host != hostName {
```

Comment:
r.Request.Host may include the port (eg 127.0.0.1:9650) which will not equal the hostName (127.0.0.1) which leads to this request 
https://docs.camino.network/developer/apis/camino-node-apis/admin#admingetnodesigner
failing 
if hostName=127.0.0.1. 
Also, what happens if the flag `--http-host` is expclitly set, yet set to 127.0.0.1 ? Does this fix address this case?

Reply:
I can split by the colon and also pass down the port to compare, so that should be fixable.
When `--http-host` is set to the loopback address it will just allow connections via the interface that this loopback is configured to bind to. Which is the indended behaviour. It becomes tricky if you want to put this behind a proxy/firewall as the proxy/firewall have to rewrite the host header. You could add an exception if a `X-Forwarded-For` header is present, but that would just shift the issue, not fix it.